src/components/App/App.test.js

```diff
@@ -0,0 +1,35 @@
+import "jsdom-global/register";
+import App from "./App";
+import React from "react";
+import { shallow } from "enzyme";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+import toJson from "enzyme-to-json";
+import { BrowserRouter as Router } from "react-router-dom";
+
+describe("App ", () => {
+  const initialState = {};
+  const mockStore = configureStore();
+
+  it("App to match snapshot", () => {
+    const store = mockStore(initialState);
+    const wrapper = shallow(
+      <Provider store={store}>
+        {/* <Router location="home"> */}
+        <App />
+        {/* </Router> */}
+      </Provider>
+    );
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it("App to match snapshot", () => {
+    UseEffect = jest.spyOn("useEffect").mockImplementation((f) => f());
```

Comment:
why useEffect is created like that?
And why would we try to mock it as empty function?
if you need some tests connected to hooks ro something like that I'd suggest using libraries
for example @testing-library/react-hooks
though you're already using enzyme, I'd google if they give something on that matter